src/protobuf-net/Meta/MetaType.cs

```diff
@@ -1615,6 +1614,35 @@ private ValueMember AddField(int fieldNumber, string memberName, Type itemType,
             return newField;
         }
 
+        private MemberInfo GetSingleMember(MemberInfo[] members)
+        {
+            if (members is null) return null;
+            if (members.Length == 1) return members[0];
+
+            MemberInfo memberFromThisType = null;
+            foreach (var member in members)
+            {
+                if (member.DeclaringType == Type)
+                {
+                    memberFromThisType = member;
+                    break;
+                }
+            }
+            // Ensure no collisions in backing fields
+            if (memberFromThisType != null && memberFromThisType.MemberType == MemberTypes.Property)
+            {
+                var prop = (PropertyInfo)memberFromThisType;
+                if (!prop.CanWrite) // otherwise we don't try to find backing field in AddField

```

Comment:
I have some concerns about this check - this code can be broken if somebody will use backing fields even if we can write directly to property and forgot to remove `if (!prop.CanWrite)` here. I add it for performance reasons in first place.
What do you think about it?